File: CS/1656_Design_an_Ordered_Stream.cs
public class OrderedStream {

    private String[] array;
    private int ptr = 0;
    public OrderedStream(int n) {
        array = new String[n];
    }
    
    public IList<string> Insert(int idKey, string value) {
        array[idKey - 1] = value;
        IList<string> result = new List<string>();
        for(int i = ptr; i < array.Length; i++){
            if(array[i] == null){
                ptr = i;
                break;
            }
            else{
                result.Add(array[i]);
            }
        }
        return result;
    }
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream obj = new OrderedStream(n);
 * IList<string> param_1 = obj.Insert(idKey,value);
 */

File: Java/1656_Design_an_Ordered_Stream.java
class OrderedStream {

    private String[] arr;
    private int ptr = 0;
    public OrderedStream(int n) {
        arr = new String[n];
    }
    
    public List<String> insert(int idKey, String value) {
        List<String> result = new ArrayList<>();
        
        arr[idKey-1] = value;
        for(int i = ptr; i < arr.length; i++){
            if(arr[i] == null){
                break;
            }
            else{
                result.add(arr[i]);
                ptr = i+1;
            }
        }
        return result;
    }
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream obj = new OrderedStream(n);
 * List<String> param_1 = obj.insert(idKey,value);
 */
